### PR TITLE
Aca 142

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ACA-142

### Change description ###
New ACA API: Get Case assigned users and roles

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by danlysiak — COMMENTED** [review]
Couple of minor initial comments

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+role.assignment.api.host=${ROLE_ASSIGNMENT_URL:http://localhost:5555}
+
+# Pooling connection manager properties for Rest Template
+http.client.max.total=${HTTP_CLIENT_MAX_TOTAL:100}
+http.client.seconds.idle.connection=${HTTP_CLIENT_SECONDS_IDLE_CONNECTION:120}
+http.client.max.client_per_route=${HTTP_CLIENT_MAX_CLIENT_PER_ROUTE:20}
+http.client.validate.after.inactivity=${HTTP_CLIENT_VALIDATE_AFTER_INACTIVITY:2000}
+http.client.connection.timeout=${HTTP_CLIENT_CONNECTION_TIMEOUT:30000}
+http.client.read.timeout=${HTTP_CLIENT_READ_TIMEOUT:60000}
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/resources/application.properties]: moved to application.yaml

File: src/main/java/uk/gov/hmcts/reform/managecase/api/payload/RoleAssignment.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private List<String> authorisations;
+    private RoleAssignmentAttributes attributes;
+
+    public boolean isNotExpiredRoleAssignment() {
+        if (beginTime != null && endTime != null) {
+            final Instant machineTimestamp = Instant.now();
+            return machineTimestamp.isAfter(beginTime) && machineTimestamp.isBefore(endTime);
+        }
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/reform/managecase/api/payload/RoleAssignment.java]: updated the code

File: build.gradle [hunk 3]
```diff
@@ -337,6 +337,10 @@ dependencies {
   implementation group: 'org.modelmapper', name: 'modelmapper', version: '2.3.7'
   implementation 'uk.gov.service.notify:notifications-java-client:3.15.1-RELEASE'
 
+  implementation 'org.projectlombok:lombok-mapstruct-binding:0.2.0'
+  annotationProcessor 'org.mapstruct:mapstruct-processor:1.3.1.Final', 'org.projectlombok:lombok:1.18.12'
+  testAnnotationProcessor 'org.mapstruct:mapstruct-processor:1.3.1.Final', 'org.projectlombok:lombok:1.18.12'
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply build.gradle]: removed

File: src/test/java/uk/gov/hmcts/reform/managecase/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations = "classpath:test.properties")
+public abstract class BaseTest {
```
Comment (danlysiak) [inline src/test/java/uk/gov/hmcts/reform/managecase/BaseTest.java:33]: This class and `WireMockBaseTest` are both for use in integration tests. In ACA we already have equivalents to these set up in `integrationTest` and `test` is just used for unit tests. Could we move the integration tests to the same place please?

Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/reform/managecase/BaseTest.java]: moved the class and deleted the classes repeated

**Review by danlysiak — APPROVED** [review]

**Review by danlysiak — APPROVED** [review]

**Comment by hemantt** [comment]
ALL SET AND READY!!

**Review by hemantt — APPROVED** [review]

**Pull request merged**